packages/components/src/Combobox/components/ComboboxContent/ComboboxContentHeader/ComboboxContentHeader.test.tsx

```diff
@@ -0,0 +1,214 @@
+            onSelectAll={onSelectAll}
+            onClearAll={onClearAll}
+            selectedCount={0}
+            searchValue="Bag"
+            options={[
+              { id: "1", label: "Bilbo Baggins" },
+              { id: "2", label: "Frodo Baggins" },
+              { id: "3", label: "Samwise Gamgee" },
+            ]}
+          />,
+        );
+
+        expect(getByText("Select all")).toBeInTheDocument();
+      });
+    });
+
+    describe("When the search term does not match any options", () => {
+      it("should not render a button in the header", () => {
+        const { queryByText } = render(
+          <ComboboxContentHeader
+            onSelectAll={onSelectAll}
+            onClearAll={onClearAll}
+            selectedCount={0}
+            searchValue="Taylor"
+            options={[
+              { id: "1", label: "Bilbo Baggins" },
+              { id: "2", label: "Frodo Baggins" },
+              { id: "3", label: "Samwise Gamgee" },
+            ]}
+          />,
+        );
+
+        expect(queryByText("Select all")).not.toBeInTheDocument();
+        expect(queryByText("Clear")).not.toBeInTheDocument();
+      });
+    });
+  });
+
+  describe("When one or more options has been selected", () => {
+    it("should render a button that is labelled 'Clear'", () => {
+      const { getByText } = render(
+        <ComboboxContentHeader
+          onSelectAll={onSelectAll}
+          onClearAll={onClearAll}
+          selectedCount={1}
+          searchValue=""
+          options={[
+            { id: "1", label: "Bilbo Baggins" },
+            { id: "2", label: "Frodo Baggins" },
+          ]}
+        />,
+      );
+
+      expect(getByText("Clear")).toBeInTheDocument();
+    });
+
+    it("should render a label '{selectedCount} selected'", () => {
+      const { getByText } = render(
+        <ComboboxContentHeader
+          onSelectAll={onSelectAll}
+          onClearAll={onClearAll}
+          selectedCount={3}
+          searchValue=""
+          options={[
+            { id: "1", label: "Bilbo Baggins" },
+            { id: "2", label: "Frodo Baggins" },
+            { id: "3", label: "Samwise Gamgee" },
+            { id: "4", label: "Meriadoc Brandybuck" },
+          ]}
+        />,
+      );
+
+      expect(getByText("3 selected")).toBeInTheDocument();
+    });
+
+    it("should call onClearAll when the 'Clear' button is clicked", () => {
+      const { getByText } = render(
+        <ComboboxContentHeader
+          onSelectAll={onSelectAll}
+          onClearAll={onClearAll}
+          selectedCount={4}
+          searchValue=""
+          options={[
+            { id: "1", label: "Bilbo Baggins" },
+            { id: "2", label: "Frodo Baggins" },
+            { id: "3", label: "Samwise Gamgee" },
+            { id: "4", label: "Meriadoc Brandybuck" },
+          ]}
+        />,
+      );
+
+      fireEvent.click(getByText("Clear"));
+
+      expect(onClearAll).toHaveBeenCalledTimes(1);
```

Comment:
so I see we have the `onClearAll` and `onSelectAll` defined globally-ish within the scope of this larger `describe` I'm guessing that's to avoid re-defining it each time, but one downside of that is that since it's in that outer scope it will "remember" how many times it has been called, and if we had any other tests in here that made assertions around calling it, those could fail

so imagine after this test we have the inverse assertion of 

`expect(onClearAll).not.toHaveBeenCalled()`

this would fail because the same instance of `onClearAll` was called in a previous test

I'd recommend being a bit more repetitive in this instance and making a fresh, locally scoped version of `onClearAll` and `onSelectAll` for each test to do its assertions against so that it doesn't trip someone up in the future

alternatively, if we want to avoid writing the same thing out over and over, we can always make a helper function that manages the setup, gives us a fresh set of actions and then we just pass in whatever is unique to our test's setup as args. in this case that might be a bit much, but up to you!